### ISSUE TITLE
"Cannot convert undefined or null to object"

### ISSUE THREAD

**Issue by tschoffelen** [mzgoddard/hard-source-webpack-plugin#282]
I'm currently getting the following error when running webpack with this plugin. Any idea what might be causing this?

```
[hard-source:core] HardSourceWebpackPlugin is writing to a new confighash path for the first time: .../web/node_modules/.cache/hard-source/87891313f4e64792ecc91301bbd04c685d7f4761cedc1408a4327c66ca5235e5
.../web/node_modules/hard-source-webpack-plugin/lib/hard-basic-dependency-plugin.js:226
    return Array.from(arg);
                 ^

TypeError: Cannot convert undefined or null to object
    at Function.from (native)
    at Object.activeExports (.../web/node_modules/hard-source-webpack-plugin/lib/hard-basic-dependency-plugin.js:226:18)
    at freezeDependency (.../web/node_modules/hard-source-webpack-plugin/lib/hard-basic-dependency-plugin.js:305:46)
    at Compiler.<anonymous> (.../web/node_modules/hard-source-webpack-plugin/lib/hard-basic-dependency-plugin.js:391:21)
    at Compiler.applyPluginsWaterfall (.../web/node_modules/tapable/lib/Tapable.js:92:24)
    at Object.exports.call (.../web/node_modules/hard-source-webpack-plugin/lib/util/plugin-compat.js:85:39)
    at freeze (.../web/node_modules/hard-source-webpack-plugin/index.js:1525:29)
    at .../web/node_modules/hard-source-webpack-plugin/index.js:1550:18
    at Array.map (<anonymous>)
    at mapMap (.../web/node_modules/hard-source-webpack-plugin/index.js:1549:22)
    at mapFreeze (.../web/node_modules/hard-source-webpack-plugin/index.js:1556:14)
    at Compiler.<anonymous> (.../web/node_modules/hard-source-webpack-plugin/lib/hard-dependency-block-plugin.js:157:21)
    at Compiler.applyPluginsWaterfall (.../web/node_modules/tapable/lib/Tapable.js:92:24)
    at Object.exports.call (.../web/node_modules/hard-source-webpack-plugin/lib/util/plugin-compat.js:85:39)
    at freeze (.../web/node_modules/hard-source-webpack-plugin/index.js:1525:29)
    at Compilation.pluginCompat.tap.module (.../web/node_modules/hard-source-webpack-plugin/lib/hard-normal-module-plugin.js:346:35)
    at Compilation.applyPlugins1 (.../web/node_modules/tapable/lib/Tapable.js:75:14)
    at module.build (.../web/node_modules/webpack/lib/Compilation.js:173:9)
    at doBuild (.../web/node_modules/webpack/lib/NormalModule.js:307:11)
    at runLoaders (.../web/node_modules/webpack/lib/NormalModule.js:209:11)
    at .../web/node_modules/loader-runner/lib/LoaderRunner.js:370:3
    at iterateNormalLoaders (.../web/node_modules/loader-runner/lib/LoaderRunner.js:211:10)
    at Array.<anonymous> (.../web/node_modules/loader-runner/lib/LoaderRunner.js:202:4)
    at Storage.finished (.../web/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:40:15)
    at .../web/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:77:9
    at .../web/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3)
```

**Comment by mzgoddard** [comment]
@tschoffelen I can see what's going on immediately from the stack trace.

`at Object.activeExports (.../web/node_modules/hard-source-webpack-plugin/lib/hard-basic-dependency-plugin.js:226:18)`

`hard-source` is freezing a the dependencies of a module. Specifically here it is freezing a `HarmonyExportImportedSpecifierDependency` which does not have a set `activeExports` member (Normally its a `Set` and needs to be turned into an `Array` or other type that'll get meaningfully serialized in JSON).

I'm not sure how you have gotten a `HarmonyExportImportedSpecifierDependency` with an unset `activeExports` ... Maybe that's a second `hard-source` bug leading to this.

**Comment by mzgoddard** [comment]
This should be fixed by #281. #281 wasn't intended for fixing this but it should have that benefit anyways.

**Comment by tschoffelen** [comment]
Yes, I can confirm that it is solved with the merge of that PR. Ended up finding that PR yesterday and merging it into my fork, and that worked like a charm. 

Thanks for the prompt response!
